File: Dominio/Entidades/Veiculo/ModeloVeicular.cs
using Dominio.Enums.Veiculo;

namespace Dominio.Entidades.Veiculo
{
    public class ModeloVeicular : BaseEntity
    {
        public  bool Situacao { get; }
        public  bool DescricaoModelo { get; }
        public  TipoModeloVeiculoEnum Tipo { get; }
        public  int QuantidadeAssento { get; }
        public  int QuantidadeEixo { get; }
        public  int CapacidadeMaxima { get; }
        public  int PassageirosEmPe { get; }
        public  bool PossuiBanheiro { get; }
        public  bool PossuiClimatizador { get; }
        public  ICollection<Veiculo> Veiculos { get; }
        public string DescricaoAtivo
        {
            get { return Situacao ? "Ativo" : "Inativo"; }
        }
    }
}

File: frontend/src/app/dominio/entidade/veiculo/modelo-veicular.ts
import { FormCampo, FiltroGrid } from '../../../services/decorator/formulario-decorator';
import { TipoModeloVeiculoEnum } from '../../enum/veiculo/tipo-modelo-veiculo-enum';

export class ModeloVeicular {
    @FormCampo('ID', 'numero', false)
    public id?: number;

    @FormCampo('Situação', 'bool', true, true)
    @FiltroGrid('Situação', 'bool', true)
    public situacao?: boolean;

    @FormCampo('Situação Descrição', 'texto', false, true)
    public situacaoDescricao?: string;

    @FormCampo('Descrição', 'texto', true, true)
    public descricao?: string;

    @FormCampo('Descrição do Modelo', 'texto', true, true)
    @FiltroGrid('Descrição do Modelo', 'texto', true)
    public descricaoModelo?: string;

    @FormCampo('Tipo', 'enum', true, false, TipoModeloVeiculoEnum)
    @FiltroGrid('Tipo', 'enum', true, TipoModeloVeiculoEnum, 'equals')
    public tipo?: TipoModeloVeiculoEnum;

    @FormCampo('Tipo Descrição', 'texto', false, true)
    public tipoDescricao?: string;

    @FormCampo('Quantidade de Assentos', 'numero', true, false)
    public quantidadeAssento?: number;

    @FormCampo('Quantidade de Eixos', 'numero', true, false)
    public quantidadeEixo?: number;

    @FormCampo('Capacidade Máxima', 'numero', true, false)
    public capacidadeMaxima?: number;

    @FormCampo('Passageiros em Pé', 'numero', true, false)
    public passageirosEmPe?: number;

    @FormCampo('Possui Banheiro', 'bool', true, false)
    public possuiBanheiro?: boolean;

    @FormCampo('Possui Banheiro Descrição', 'texto', false, true)
    public possuiBanheiroDescricao?: string;

    @FormCampo('Possui Climatizador', 'bool', true, false)
    public possuiClimatizador?: boolean;

    @FormCampo('Possui Climatizador Descrição', 'texto', false, true)
    public possuiClimatizadorDescricao?: string;

    @FormCampo('Data Criação', 'texto', false, true)
    public createdAt?: Date;

    @FormCampo('Data Atualização', 'texto', false, true)
    public updatedAt?: Date;
}
